### DISCUSSION TITLE
Empty allure report considered successful

### DISCUSSION THREAD

**Discussion by vladbeza** [allure-framework/allure2#1357] [General Discussion]
I use allure plugin to generate and post report on jenkins. When there are no tests in report for some reason (most commonly something went wrong), empty report is generated but build is green. I want to get failed build in this case. How can I achieve this?

**Comment by Avaddon-tfa** [comment]
Hello there! 

In this case, the pipeline has to stay green by design. The main responsibility of the Allure Report is to make reports, not to manage the pipeline state.

At the moment we are not planning to implement the feature you've requested for at least three reasons:
- any other reporting tool builds reports regardless of tests presence or number;
- testing frameworks have their own logic and tooling to check if the tests have been run or not;
- within CI (e.g. Jenkins or Gitlab) logic, there are body scripts (breaking a pipeline at this point is fine) and post-job scripts (breaking a pipeline at this point is bad). Reporting tools are often used in the second stage. So throwing an exception would break a lot of pipelines and would work out of the CI rules.

**Reply by vladbeza** [reply]
It is clear for now. Thank you for response!
